drivers/accel/adxl355/adxl355.c

```diff
@@ -912,8 +942,16 @@ static int64_t adxl355_accel_conv(struct adxl355_dev *dev,
 		accel_data = raw_accel;
 
 	// Apply scale factor based on the selected range
-	return ((int64_t)(accel_data * ADXL355_ACC_SCALE_FACTOR_MUL *
-			  adxl355_scale_mul[dev->range]));
+	switch (dev->dev_type) {
+	case ID_ADXL355:
+		return ((int64_t)(accel_data * ADXL355_ACC_SCALE_FACTOR_MUL *
+				  adxl355_scale_mul[dev->range]));
+	case ID_ADXL359:
+		return ((int64_t)(accel_data * ADXL359_ACC_SCALE_FACTOR_MUL *
+				  adxl355_scale_mul[dev->range]));
+	default:
+		return 0;
```

Comment:
`-EINVAL` ?

Reply:
in this case the returned value can also be negative. the default case will never be reached. 0 return value in case of an error is suitable in this case.